cmd/kueuectl/app/util/completion.go

```diff
@@ -0,0 +1,204 @@
+	}
+}
+
+func ContextsCompletionFunc(clientGetter ClientGetter) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		config, err := clientGetter.ToRawKubeConfigLoader().RawConfig()
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+		var validArgs []string
+		for name := range config.Contexts {
+			if strings.HasPrefix(name, toComplete) {
+				validArgs = append(validArgs, name)
+			}
+		}
+		return validArgs, cobra.ShellCompDirectiveNoFileComp
+	}
+}
+
+func ClustersCompletionFunc(clientGetter ClientGetter) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		config, err := clientGetter.ToRawKubeConfigLoader().RawConfig()
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+		var validArgs []string
+		for name := range config.Clusters {
+			if strings.HasPrefix(name, toComplete) {
+				validArgs = append(validArgs, name)
+			}
+		}
+		return validArgs, cobra.ShellCompDirectiveNoFileComp
+	}
+}
+
+func UsersCompletionFunc(clientGetter ClientGetter) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		config, err := clientGetter.ToRawKubeConfigLoader().RawConfig()
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+		var validArgs []string
+		for name := range config.AuthInfos {
+			if strings.HasPrefix(name, toComplete) {
+				validArgs = append(validArgs, name)
+			}
+		}
+		return validArgs, cobra.ShellCompDirectiveNoFileComp
+	}
+}
+
+func WorkloadNameCompletionFunc(clientGetter ClientGetter, status *bool) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		if len(args) > 0 {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+
+		clientSet, err := clientGetter.KueueClientSet()
+		if err != nil {
+			return []string{}, cobra.ShellCompDirectiveError
+		}
+
+		namespace, _, err := clientGetter.ToRawKubeConfigLoader().Namespace()
+		if err != nil {
+			return []string{}, cobra.ShellCompDirectiveError
+		}
+
+		list, err := clientSet.KueueV1beta1().Workloads(namespace).List(cmd.Context(), metav1.ListOptions{})
+		if err != nil {
+			return []string{}, cobra.ShellCompDirectiveError
+		}
+
+		if status != nil {
+			filteredItems := make([]v1beta1.Workload, 0, len(list.Items))
+			for _, wl := range list.Items {
+				if ptr.Deref(wl.Spec.Active, true) == *status {
+					filteredItems = append(filteredItems, wl)
+				}
+			}
+			list.Items = filteredItems
+		}
+
+		validArgs := make([]string, len(list.Items))
+		for i, wl := range list.Items {
```

Comment:
same question whether you need `if strings.HasPrefix(name, toComplete)`

And same for the rest of the list calls